python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -314,50 +328,145 @@ cdef class Column:
+            Must implement the `__array_interface__` protocol.
+
+        Raises
+        ------
+        NotImplementedError
+            This method is not yet implemented.
+        """
+        raise NotImplementedError(
+            "Converting to a pylibcudf Column is not yet implemented."
+        )
+
+    @classmethod
+    def from_cuda_array_interface(cls, obj):
+        """
+        Create a Column from an object implementing the CUDA Array Interface.
 
         Parameters
         ----------
         obj : object
-            The object with the CUDA array interface to create a column from.
+            Must implement the `__cuda_array_interface__` protocol.
 
         Returns
         -------
         Column
             A Column containing the data from the CUDA array interface.
 
-        Notes
-        -----
-        Data is not copied when creating the column. The caller is
-        responsible for ensuring the data is not mutated unexpectedly while the
-        column is in use.
+        Raises
+        ------
+        ValueError
+            If the object is not 1D or 2D, or is not C-contiguous.
+        ImportError
+            If CuPy is not installed.
         """
-        data = gpumemoryview(obj)
-        iface = data.__cuda_array_interface__
-        if iface.get('mask') is not None:
-            raise ValueError("mask not yet supported.")
+        iface = getattr(obj, "__cuda_array_interface__", None)
+        if iface is None:
+            raise TypeError("Object does not implement __cuda_array_interface__")
 
-        typestr = iface['typestr'][1:]
+        if iface.get("mask") is not None:
+            raise ValueError("mask not yet supported")
+
+        typestr = iface["typestr"][1:]
         data_type = _datatype_from_dtype_desc(typestr)
 
-        if not is_c_contiguous(
-            iface['shape'],
-            iface['strides'],
-            size_of(data_type)
-        ):
+        shape = iface["shape"]
+        if not is_c_contiguous(shape, iface["strides"], size_of(data_type)):
             raise ValueError("Data must be C-contiguous")
 
-        size = iface['shape'][0]
-        return Column(
-            data_type,
-            size,
-            data,
-            None,
-            0,
-            0,
-            []
+        if len(shape) == 1:
+            data = gpumemoryview(obj)
+            size = shape[0]
+            return cls(data_type, size, data, None, 0, 0, [])
+        elif len(shape) == 2:
+            if np is None or cp is None:
+                raise ImportError(
+                    "NumPy and CuPy must be installed to use this method on 2D data"
+                )
+            flat_data = cp.asarray(obj).ravel()
+            num_rows, num_cols = shape
+            if num_rows < INT_MAX:
+                offsets_col = sequence(
+                    num_rows + 1,
+                    Scalar.from_numpy(np.int32(0)),
+                    Scalar.from_numpy(np.int32(num_cols))
```

Comment:
*nit*: This should be `size_type`, but we shouldn't introduce the mapping.

What I would like (followup) is a `Scalar.from_py(value, dtype)` so you can specify the dtype when coming from python.

Reply:
Will do